NEWS.md

```diff
@@ -1,3 +1,13 @@
+# DataFrames v1.0.x Patch Release Notes
+
+* patch v1.0.3: make sure `subset` checks if the passed condition function
+  returns a vector of values
+  ([#XXXX](https://github.com/JuliaData/DataFrames.jl/pull/XXXX))
```

Comment:
The question is how we qualify this change. Since we added this feature in 1.0 release only I would prefer to say this is a bug (i.e. not intended functionality) so we patch it. Otherwise we should deprecate it and it could be changed only in 2.0 release.

Reply:
Yeah in theory no breaking changes should be introduced until 2.0. In practice there are lots of grey areas like this one. Just like Julia allows "minor changes" across minor versions, I think we can do this kind of thing given that the risk of breakage is limited. But this PR doesn't fix any obviously incorrect behavior nor allows doing something that currently fails -- it's just an improvement to limit confusion. Better keep disruption in patch releases to the strict minimum IMO.

Reply:
So in summary do we go for 1.1.0 or 1.0.3 release with this? I understand you feel 1.1.0 would be better. Right? I am OK with this (I guess this should not cause package update issues).

Reply:
Right, I think 1.1.0 would be better. Not sure whether it's simpler to merge the PR and create a release-1.1 branch, or whether we should keep it open for a while.

Reply:
I am OK to switch to 1.1.0 but I would make the release as soon as possible. I do not want people using the functionality we want to remove.

Reply:
OK. I admit it's a bit weird to tag 1.1 so quickly, but in practice that's not a problem.

Reply:
Yes - it is weird, but it is not a problem so I would go for it (and explain it in my blog on Friday 😄).